api/health_check.proto

```diff
@@ -88,6 +88,14 @@ message HealthCheck {
   message RedisHealthCheck {
   }
 
+  // [#not-implemented-hide:] grpc.health.v1.Health-based healthcheck.
+  // See `GRPC doc <https://github.com/grpc/grpc/blob/master/doc/health-checking.md>`
+  // for details.
+  message GrpcHealthCheck {
+    // Name of the service to perform check on.
+    string service = 1;
+  }
```

Comment:
How do we send a health check request to a gRPC service? What the request looks like? What is the response would be? How does the client knows whether to use mTLS to send the request or something else?

Reply:
I'm not sure if I understand the question correctly. [Standard gRPC health protocol](https://github.com/grpc/grpc/blob/master/src/proto/grpc/health/v1/health.proto) will be used, it was [chosen here](https://github.com/envoyproxy/envoy/issues/369#issuecomment-311542604). As for mTLS, I don't think gRPC healthcheck differs from usual HTTP one - all transport layer security stuff is defined in `tls_context` of the cluster being health-checked.

Reply:
Yup exactly. Cluster connection options are used.